src/client/api/pythonApi.ts

```diff
@@ -107,7 +115,7 @@ export class PythonExtensionChecker implements IPythonExtensionChecker {
         return this.extensions.getExtension(this.pythonExtensionId) !== undefined;
     }
 
-    public async showPythonExtensionInstallRequiredPrompt(): Promise<void> {
+    public async showPythonExtensionInstallRequiredPrompt(): Promise<boolean | void> {
```

Comment:
It's not immediately obvious to me what a boolean or void return value means for this function. A comment would be nice, or better yet an enum if these three states are distinct.

Reply:
We don't need void, we update `waitingOnInstallPrompt` to contain the result of the installation. 
Then we'd only have a boolean, though I'd still prefer the enum (we can use InstallerResponse)